app/routes/stakeholder-router.js

```diff
@@ -1,15 +1,28 @@
 const router = require("express").Router();
 const stakeholderController = require("../controllers/stakeholder-controller");
+const jwtSession = require("../../middleware/jwt-session");
 
-router.get("/", stakeholderController.search);
-router.get("/dashboard", stakeholderController.searchDashboard);
-router.get("/:id", stakeholderController.getById);
-router.post("/csv", stakeholderController.csv);
-router.post("/", stakeholderController.post);
-router.put("/:id", stakeholderController.put);
-router.delete("/:id", stakeholderController.remove);
-router.put("/:id/needsVerification", stakeholderController.needsVerification);
-router.put("/:id/assign", stakeholderController.assign);
-router.put("/:id/claim", stakeholderController.claim);
+router.get("/", jwtSession.validateUser, stakeholderController.search);
```

Comment:
to ensure authenticated users can hit the `search` endpoint we just add the `jwtSession.validateUser` method.

Reply:
This one particular endpoint needs to be used by unauthenticated users. That being said, we are basically giving away all our hard-earned data via an open endpoint. Maybe we should think about at least encrypting the response data, so it isn't quite as easy for anybody to take all our data directly from the JSON result to re-publish as their own?

Reply:
> That being said, we are basically giving away all our hard-earned data via an open endpoint. Maybe we should think about at least encrypting the response data,

I can't think of a way to keep as you say someone from  as you taking all the data from the JSON _and_ letting the API have unauthenticated access. 

even if the data is encrypted the keys would have to be baked into the client code so that it could decrypt the riesponse -- and if the keys are in client they are visible to the world via browser dev tools.

Reply:
I think if the API is open it should be open and that's good and ok. What's the worst case scenario? 

If an evil corporation scrapes all food-oasis APIs and makes 1 million $ on a food source app then that just means there's _more_ resources for food seekers! If you slap a GPL license in there, then let the lawyers to figure it out :smile:

Reply:
Sounds right to me.